src/state/books/BooksContext.tsx

```diff
@@ -0,0 +1,46 @@
+import React, {createContext, useReducer, ReactNode} from 'react';
+import BooksReducer from './BooksReducer';
+import { IAddBookAction, IClearBooksAction } from './actionTypes';
+import {BooksActionTypes} from '../books/actionTypes';
+import { Book } from '../../models/Book';
+
+export interface BookState{
+    books: Book[],
+    addBook: (book: Book) => void,
+    clearBooks: () => void
+}
+
+export const initialState: BookState = {
+    books: [],
+    addBook: (book: Book) => {},
```

Comment:
Consider having this handle books, not just a book. Looks like you never add just one book, so you might as well have it handle a list of books 🤷‍♂️